### DISCUSSION TITLE
Settlement Report ERROR  'message': 'Access to requested resource is denied.', 'code': 'Unauthorized'

### DISCUSSION THREAD

**Discussion by Noorilahi** [amzn/selling-partner-api-models#3144] [General] [answered]
Order Api and Other Finance Api are Working fine and in Report API Some Report are generating fine but when i try for Settlement Report getting

Fun Call:
report_type='GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE'
start_date = (datetime.utcnow() - timedelta(days=30)).isoformat()
end_date = (datetime.utcnow() - timedelta(days=0)).isoformat()
res = Reports(credentials=credentials, marketplace=Marketplaces.IN)
res.get_reports(reportType=report_type, processingStatuses="DONE", marketplaceIds=Marketplaces.IN.marketplace_id, pageSize=1, createdSince=start_date, createdUntil=end_date)

ERROR "

Exception Type: SellingApiForbiddenException
Exception Value: [{'message': 'Access to requested resource is denied.', 'code': 'Unauthorized', 'details': ''}]
Exception Location: /home/user/.local/lib/python3.8/site-packages/sp_api/base/client.py, line 112, in _check_response
"

Thanks

**Comment by ezpinsky** [comment, marked as answer]
The settlement report can't be created by using the createReport operation. It is auto generated by Amazon for data within the last 90 days and each report generated contains 2 weeks of data. To access these generated reports you need to hit the [getReports](https://github.com/amzn/selling-partner-api-docs/blob/main/references/reports-api/reports_2020-09-04.md#getreports) operation in the reports api and filter for the settlement report. Be sure to notice that this operation does not use startDate and endDate but createdUntil and createdSince. Default time interval is 90 days back I think. Anyway, the response will have the report ids of the pre generated settlement reports. Just save the report ids from the response and now you can use your normal report API flow to download the report. Hope this helps!

**Comment by Noorilahi** [comment]
Hi @ezpinsky,

Thank you,
It worked for me
